plugin/storage/es/spanstore/writer.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package spanstore
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/uber/jaeger/model"
+	"github.com/uber/jaeger/model/converter/json"
+	jModel "github.com/uber/jaeger/model/json"
+	"github.com/uber/jaeger/pkg/es"
+)
+
+const spanType = "span"
+const serviceType = "service"
+const hostPort = "http://localhost:9200"
+
+// SpanWriter is a wrapper around elastic.Client
+type SpanWriter struct {
+	client es.Client
+	logger *zap.Logger
+}
+
+// Service is the JSON struct for service:operation documents in ElasticSearch
+type Service struct {
+	serviceName   string
+	operationName string
+}
+
+// NewSpanWriter creates a new SpanWriter for use
+func NewSpanWriter(client es.Client, logger *zap.Logger) *SpanWriter {
+	return &SpanWriter{
+		client: client,
+		logger: logger,
+	}
+}
+
+// WriteSpan writes a span and its corresponding service:operation in ElasticSearch
+func (s *SpanWriter) WriteSpan(span *model.Span) error {
+	ctx := context.Background()
+	jaegerIndexName := spanIndexName(span)
+	// Convert model.Span into json.Span
+	jsonSpan := json.FromDomainEmbedProcess(span)
+
+	if err := s.checkAndCreateIndex(ctx, jaegerIndexName, jsonSpan); err != nil {
+		return err
+	}
+	if err := s.writeService(ctx, jaegerIndexName, jsonSpan); err != nil {
```

Comment:
Sweet, also - it seems we're just doing {serviceName}|{operationName}

Is that the best way to do this? or should we have a documentation containing serviceName and operationName fields?